jwst/cube_build/blot_cube_build.py

```diff
@@ -182,82 +176,45 @@ def blot_images_miri(self):
             x_cube = np.ndarray.flatten(x_cube)
             y_cube = np.ndarray.flatten(y_cube)
             flux_cube = np.ndarray.flatten(self.cube_flux)
+
             valid = ~np.isnan(y_cube)
-            x_cube = x_cube[valid]
-            y_cube = y_cube[valid]
-            flux_cube = flux_cube[valid]
             valid_channel = np.logical_and(x_cube >= xstart, x_cube <= xend)
-            x_cube = x_cube[valid_channel]
-            y_cube = y_cube[valid_channel]
-            flux_cube = flux_cube[valid_channel]
-            # ___________________________________________________________________
+            valid_flux = (flux_cube != 0)
+
+            fuse = np.where(valid & valid_channel & valid_flux)
+            x_cube = x_cube[fuse]
+            y_cube = y_cube[fuse]
+            flux_cube = flux_cube[fuse]
+
             log.info('Blotting back to %s', model.meta.filename)
             # ______________________________________________________________________________
-            # For every detector pixel find the overlapping median cube spaxels.
+            # blot_wrapper is a c extension that finds:
+            # the overlapping median cube spaxels with the detector pixels
             # A median spaxel that falls withing the ROI of the center of the
-            # detector pixel in the tangent plane is flagged as an overlapping
-            # pixel.
-            # the Regular grid is on the x,y detector
+            # detector pixel  is flagged as an overlapping pixel.
+
+            xsize2 = xcenter.shape[0]
+            roi_det = 1.0  # Just large enough that we don't get holes
+
+            t0 = time.time()
```

Comment:
Either delete this or move it to the top of the function to provide just one measurement of execution time. That is just a suggestion but I think numerous logging messages with timings that few people know what they are measuring is just going to overwhelm readers of the log files.

Reply:
I will just delete it. I was using it to see how much the speed improved. Not needed now